controller/hydrator_dependencies.go

```diff
@@ -54,6 +54,15 @@ func (ctrl *ApplicationController) GetRepoObjs(origApp *appv1.Application, drySo
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to get repo objects: %w", err)
 	}
+	trackingMethod, err := ctrl.settingsMgr.GetTrackingMethod()
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get tracking method: %w", err)
+	}
+	for _, obj := range objs {
+		if err := argoutil.NewResourceTracking().RemoveAppInstance(obj, trackingMethod); err != nil {
+			return nil, nil, fmt.Errorf("failed to remove the app instance value: %w", err)
+		}
+	}
```

Comment:
Is there any chance we can test GetRepoObjs for this behavior? The ResourceTracking unit tests are awesome, but I think we could use something at a higher level too.